pkg/strategy/grid2/strategy.go

```diff
@@ -761,25 +764,29 @@ func (s *Strategy) calculateBaseQuoteInvestmentQuantity(quoteInvestment, baseInv
 	// maxBaseQuantity = baseInvestment / numberOfSellOrders
 	// if maxBaseQuantity < minQuantity or maxBaseQuantity * priceLowest < minNotional
 	// then reduce the numberOfSellOrders
-	numberOfSellOrders := 0
-	for i := len(pins) - 1; i >= 0; i-- {
-		pin := pins[i]
-		price := fixedpoint.Value(pin)
-		sellPrice := price
-		if s.ProfitSpread.Sign() > 0 {
-			sellPrice = sellPrice.Add(s.ProfitSpread)
-		}
+	numberOfSellOrders := s.BaseGridNum
+
+	// if it's not configured
+	if numberOfSellOrders == 0 {
+		for i := len(pins) - 1; i >= 0; i-- {
+			pin := pins[i]
+			price := fixedpoint.Value(pin)
+			sellPrice := price
+			if s.ProfitSpread.Sign() > 0 {
+				sellPrice = sellPrice.Add(s.ProfitSpread)
+			}
 
-		if sellPrice.Compare(lastPrice) < 0 {
-			break
-		}
+			if sellPrice.Compare(lastPrice) < 0 {
```

Comment:
Might be worth adding testcase for `sellPrice == lastPrice`.

Reply:
I can add it in the next PR